File: TEAM A/Week 7/Eren/118666. 성격 유형 검사하기/성격 유형 검사하기.cpp
#include <string>
#include <vector>

using namespace std;

int score[4];
string solution(vector<string> survey, vector<int> choices) {
    string answer = "";
    for (int i = 0; i < survey.size(); i++) {
        if (survey[i] == "RT") {
            int s = choices[i] - 4;
            score[0] += s;
        }
        else if (survey[i] == "TR") {
            int s = -(choices[i] - 4);
            score[0] += s;
        }
        else if (survey[i] == "CF") {
            int s = choices[i] - 4;
            score[1] += s;
        }
        else if (survey[i] == "FC") {
            int s = -(choices[i] - 4);
            score[1] += s;
        }
        else if (survey[i] == "JM") {
            int s = choices[i] - 4;
            score[2] += s;
        }
        else if (survey[i] == "MJ") {
            int s = -(choices[i] - 4);
            score[2] += s;
        }
        else if (survey[i] == "AN") {
            int s = choices[i] - 4;
            score[3] += s;
        }
        else if (survey[i] == "NA") {
            int s = -(choices[i] - 4);
            score[3] += s;
        }
    }
    if (score[0] > 0) { answer.push_back('T'); }
    else answer.push_back('R');
    if (score[1] > 0) { answer.push_back('F'); }
    else answer.push_back('C');
    if (score[2] > 0) { answer.push_back('M'); }
    else answer.push_back('J');
    if (score[3] > 0) { answer.push_back('N'); }
    else answer.push_back('A');
    return answer;
}

File: Team C (Season 2)/Week 3/Greed/118666. 성격 유형 검사하기/성격 유형 검사하기.py
def solution(survey, choices):
    characters = ["R","T","C","F","J","M","A","N"]
    score = [0]*len(characters)
    answer = []
    for i in range(len(survey)):
        if choices[i] <4:
            for j in range(len(characters)):
                if characters[j] == survey[i][:1]:
                    score[j] += 4 - choices[i] 
        else:
            for j in range(len(characters)):
                if characters[j] == survey[i][1:]:
                    score[j] += choices[i] - 4 

    for i in range(0,len(characters),2):
        if score[i] > score[i+1]:
            answer.append(characters[i])
        elif score[i] < score[i+1]:
            answer.append(characters[i+1])
        else:
            answer.append(characters[i] if characters[i] < characters[i+1] else characters[i+1])
    print(answer)
    return "".join(answer)